server/middleware/fetchLicenceMiddleware.ts

```diff
@@ -1,16 +1,43 @@
 import { RequestHandler } from 'express'
+import getUrlAccessByStatus from '../utils/urlAccessByStatus'
 import LicenceService from '../services/licenceService'
+import logger from '../../logger'
 
 export default function fetchLicence(licenceService: LicenceService): RequestHandler {
   return async (req, res, next) => {
     if (req.params.licenceId) {
       const { user } = res.locals
       const licence = await licenceService.getLicence(req.params.licenceId, user)
 
+      // Does this prison user have a caseload which allows access this licence?
+      if (licence && user?.nomisStaffId) {
+        if (!user.prisonCaseload.includes(licence.prisonCode)) {
+          logger.info(
+            `Prison caseload ${user?.prisonCasload} denied access to licence ID ${licence.id} ${licence.prisonCode}`
+          )
+          return res.redirect('/access-denied')
```

Comment:
Do we want to redirect or just render a different view? Not sure, redirect is probably ok

Reply:
Yea, same pattern as `/authError`, fine